### DISCUSSION TITLE
Is it a good idea to have different cmdlet names in the V2 SDK?

### DISCUSSION THREAD

**Discussion by 12Knocksinna** [microsoftgraph/msgraph-sdk-powershell#1705] [General]
I don't think so, and the precedent of having the same cmdlet names in production and beta modules was set by Azure AD and AzureADPreview. All I see is added complexity for developers to decide what module to use...

It would be nice if the API developers fixed all the bugs in the V1.0 user and groups cmdlets as that might mean that we could stay with the one V1.0 module for production scripts...

**Comment by 12Knocksinna** [comment]
My other public comments are at https://office365itpros.com/2022/12/20/microsoft-graph-powershell-sdk-v2/

**Reply by maisarissi** [reply]
To add to the discussion, [here](https://aka.ms/preview-psv2) is the PowerShell SDK public preview blog where we go through the changes, including the namespace and cmdlets names changes.

**Comment by dsolodow** [comment]
I'm not thrilled with different cmdlet names, but it seems like from an architectural perspective, splitting the modules makes a lot of sense. 
Honestly, the AzureAD/AzureADPreview ones are a good example of why **not** to use the same names. I always found it to be a massive PITA due to module autoload as it wasn't always predictable which module would load first, and then you had to unload/reload for the desired one.
